File: Lab2/src/KI/Tymkiv/Lab2/Thermostat.java
package KI.Tymkiv.Lab2;

/**
 * Клас {@code Thermostat} представляє термостат, який зберігає та управляє температурою.
 */
public class Thermostat {
    private double temperature;

    /**
     * Створює новий термостат з заданою температурою.
     *
     * @param temperature температура термостата
     */
    public Thermostat(double temperature) {
        this.temperature = temperature;
    }

    /**
     * Створює новий термостат з початковою температурою 0 градусів.
     */
    public Thermostat() {
        this.temperature = 0;
    }

    /**
     * Повертає поточну температуру термостата.
     *
     * @return температура термостата
     */
    public double getTemperature() {
        return temperature;
    }

    /**
     * Встановлює нову температуру термостата.
     *
     * @param temperature нова температура термостата
     */
    public void setTemperature(double temperature) {
        this.temperature = temperature;
    }
}

File: Thermostat.py
class Thermostat:
    """Клас Thermostat представляє термостат, який зберігає та управляє температурою."""
    
    def __init__(self, temperature=0):
        """Створює новий термостат із заданою температурою.
        
        Аргументи:
            temperature (float): температура термостата (за замовчуванням: 0)
        """
        self.temperature = temperature
    
    def get_temperature(self):
        """Повертає поточну температуру термостата.
        
        Повертає:
            float: температура термостата
        """
        return self.temperature
    
    def set_temperature(self, temperature):
        """Встановлює нову температуру термостата.
        
        Аргументи:
            temperature (float): нова температура термостата
        """
        self.temperature = temperature

